core/modules/system/config/system.core.json

```diff
@@ -15,6 +15,7 @@
       "strip_sensitive_headers_on_host_change": true,
       "strip_sensitive_headers_on_https_downgrade": true
     },
+    "backup_limit": 5,
```

Comment:
I wonder, if this "backup_limit" belongs in config.

The directory's set in settings(.local).php - probably so it's available even with weirdest failures.
Would anyone expect the limit to be in system.core.json? To me it appears to be sort of an "alien" here. I know, we can rely on config to be available at the point, this settings needed. It's actually for other reasons, I wonder, if it even belongs here.

Reply:
I considered `system.backup.json` as an alternative. Other existing files like `system.mail.json` and `system.authorize.json` are similarly separate and very slim config files. Do you think that would be better?

Reply:
How to deal with this (config) item probably depends on future steps... :thinking: 

If the plan is to provide a UI for settings in core, then a dedicated config json file makes more sense than stuffing it into system.core.json. The UI should be a topic for a follow-up, though.

If the plan is to leave this entirely to contrib/custom (to extend the class and provide more settings), then this single config item makes no sense at all.

Another thing to consider: only a handful of people will even be aware that `backup_limit` could be overridden by fiddling in JSON. You, me, the core committer who merges this PR...
It's rather pointless to provide a config item without any UI - a secret feature nobody's aware of is no feature. We could stick with a sane default (5) in that case. :stuck_out_tongue_winking_eye: Eventually with a (documented) settings.php override.

Reply:
Backup and Migrate has 28 global settings and ~20 options per advanced backup. I initially figured there would be more settings, but most of the options in B&M seem to cover features we haven't included in core (like a cron-based backup) or unnecessarily detailed.

> If the plan is to leave this entirely to contrib/custom (to extend the class and provide more settings), then this single config item makes no sense at all.

I think we'll probably see some of both. We could make an option for downloading/uploading backups for example, or cron backups, in core... it depends on where the demand and comfort level is. So maybe splitting out (even for a single value) might be good future-proofing.

Even with no UI, I feel like the value should be at least _possible_ to override, and it makes more sense as config than a hard-coded setting. It's also easier to test as config.

Reply:
So, the plan seems to be to add *some* settings via UI. Of course, never all the whistles and bells that B&M provides. The backup_limit, the file name pattern (eventually), provide a download (eventually)... all to be determined in a follow-up.

But even if there would only be three settings, I'd recommend a dedicated json file, not reusing system.core.

Reply:
Separated into `system.backup.json`, thanks for your input!